dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -225,24 +242,167 @@ class TypedDataset[T](val dataset: Dataset[T])(implicit val encoder: TypedEncode
 
     val joinedDs = FramelessInternals.mkDataset(dataset.sqlContext, joinedPlan, TypedExpressionEncoder[(T, Option[A])])
 
-    new TypedDataset[(T, Option[A])](joinedDs)
+    TypedDataset.create[(T, Option[A])](joinedDs)
   }
 
+  import scala.reflect.runtime.universe.TypeTag
+
+  def makeUDF[A: TypeTag, R: TypeTag](f: A => R)(implicit aen: TypedEncoder[A], ren: TypedEncoder[R]): TypedColumn[T, A] => TypedColumn[T, R] = functions.udfs.udf[T, A, R](f)
```

Comment:
@OlivierBlanvillain the makeUDF essentially lifts a UDF to the context of the enclosing TypedDatased.